Us b implementing booking session as user button and functionality

andreas-laine opened the pull request:
- Added Javascript for booking a session.
- Added CSS for the booking button.

andreas-laine committed:
Implement session booking functionality and enhance button styles

- Added `bookSession` and `getSession` functions to the API service for handling session bookings.
- Updated `SessionsPageUser` component to manage booking state and display appropriate messages.
- Introduced a new button style for booking sessions and adjusted card expansion behavior in CSS.

asjo-89 commented on GymSlay/src/services/api.js:
Jag är lite osäker på fetchURL:en här i booksessions. Ser det ut så i controllern? Känns som att id borde vara efter book i urlen?

asjo-89 committed:
Refactor booking logic in SessionsPageUser and update API endpoint for session booking.
Changed booked to isBooked and updated handleBook to use local parameter instead of global.

asjo-89 reviewed (approved):
I think it will work after our changes. We  have to test it after login function is completed.
